src/storage/page_buffer.c

```diff
@@ -8412,14 +8421,39 @@ pgbuf_request_data_page_from_page_server (const VPID * vpid, log_lsa target_repl
     }
   else
     {
+      PGLENGTH data_length = 0;
+      std::memcpy (&data_length, message_buf, sizeof (PGLENGTH));
+      message_buf += sizeof (PGLENGTH);
+
+      const char *decompressed_data = nullptr;
+      int decompressed_data_size = 0;
+
+      // decompress data that is sent over; there are 3 cases:
+      //  - compression setting on and data compressed successfully
+      //  - compression setting on but data failled to be compressed
+      //  - compression off
+      // all are handled in the decompress function
+      const bool decompressed = pgbuf_decompress_page (thread_r, reinterpret_cast < const void *>(message_buf),
+						       (int) data_length, decompressed_data, decompressed_data_size);
+      assert (decompressed);
+      message_buf += data_length;
+
+      if (!decompressed)
+	{
+	  // TODO: error
+	  assert_release (false);
```

Comment:
If it fails, it's catastrophic. Examples:
- if source of request is a client transaction (ATS, PTS): abort transaction
- if source of request is vacuum thread (ATS): it could retry request
- if source of request is replication thread (PTS): it could retry request
- ...

For now I think we should just crash.
I will add an error as `TODO`-ed.

Reply:
Added error message